src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1618,14 +1655,25 @@ namespace winrt::TerminalApp::implementation
     // - Responds to the TabView control's Tab Closing event by removing
     //      the indicated tab from the set and focusing another one.
     //      The event is cancelled so App maintains control over the
-    //      items in the tabview.
+    //      items in the tabview. If there are more the one pane in the focused
+    //      tab, it will show a warning
     // Arguments:
     // - sender: the control that originated this event
     // - eventArgs: the event's constituent arguments
     void TerminalPage::_OnTabCloseRequested(const IInspectable& /*sender*/, const MUX::Controls::TabViewTabCloseRequestedEventArgs& eventArgs)
     {
-        const auto tabViewItem = eventArgs.Tab();
-        _RemoveTabViewItem(tabViewItem);
+        if (auto index{ _GetFocusedTabIndex() })
+        {
+            auto focusedTab{ _GetStrongTabImpl(*index) };
+            if (focusedTab->_GetLeafPaneCount() == 1)
+            {
+                const auto tabViewItem = eventArgs.Tab();
+                _RemoveTabViewItem(tabViewItem);
```

Comment:
Should these two lines just be a single call to `_CloseFocusedTab`

Reply:
Thank you. Resolved.

Reply:
Actually, that will come up with an error.